OSDOCS-14402:YAML parameter additions (#92896)
openshift/openshift-docs: modules/microshift-config-parameters-table.adoc

|`tls.minVersion`
|`VersionTLS12` or `VersionTLS13`

> |Specifies the minimum version of TLS to serve from the API server. Default is value is `VersionTLS12`. TLS 1.3 ciphers are preset and not configurable.

Reviewer: ```suggestion
|Specifies the minimum version of TLS to serve from the API server. Default value is `VersionTLS12`. TLS 1.3 ciphers are preset and not configurable.
```

Reviewer: "TLS 1.3 ciphers are preset and not configurable". This is a confusing statement, the actual meaning is not clear?

|`number`
|By default, `etcd` uses as much memory as needed to handle the load on the system. However, in memory constrained systems, it might be preferred or necessary to limit the amount of memory `etcd` can to use at a given time.

|`ingress.certificate.Secret`

> |`string`

Reviewer 2: better to remove the last "." from "ingress.certificate.Secret", so it should be `ingress.certificateSecret`.

> |A string that is a reference to a secret that contains the default certificate that is served by the ingress controller. When Routes do not specify their own certificate, `certificateSecret` is used.

Reviewer 3: ```suggestion
|A string that is a reference to a secret that contains the default certificate that is served by the ingress controller. When routes do not specify their own certificate, `certificateSecret` is used.
```

The secret must contain the following keys and data:

* `tls.crt`: certificate file contents
* `tls.key`: key file contents

> If you do not set one, a wildcard certificate is automatically generated and used. The certificate is valid for the ingress controller `domain` and `subdomains`, and the generated certificate's CA is automatically integrated with the cluster's truststore.

Reviewer 3: ```suggestion
If you do not set one of these values, a wildcard certificate is automatically generated and used. The certificate is valid for the ingress controller `domain` and `subdomains`, and the generated certificate's CA is automatically integrated with the cluster's truststore.
```

Any certificate in use is automatically integrated in the {microshift-short} OAth server.

|`ingress.clientTLS`
|`spec.clientTLS.clientCertificatePolicy`, `spec.clientTLS.ClientCA`, `AllowedSubjectPatterns`

> |`clientTLS` authenticates client access to the cluster and services; as a result, mutual TLS authentication is enabled. If you do not set one, then client TLS is not enabled. `clientTLS` has the required subfields, `spec.clientTLS.clientCertificatePolicy` and `spec.clientTLS.ClientCA`.

Reviewer 4: > If you do not set one

One what?

> |`clientTLS` authenticates client access to the cluster and services; as a result, mutual TLS authentication is enabled. If you do not set one, then client TLS is not enabled. `clientTLS` has the required subfields, `spec.clientTLS.clientCertificatePolicy` and `spec.clientTLS.ClientCA`.

Reviewer 3: ```suggestion
|`clientTLS` authenticates client access to the cluster and services; as a result, mutual TLS authentication is enabled. If you do not set a value, the client TLS is not enabled. The `clientTLS` parameter has the required subfields, `spec.clientTLS.clientCertificatePolicy` and `spec.clientTLS.ClientCA`.
```

> |`clientTLS` authenticates client access to the cluster and services; as a result, mutual TLS authentication is enabled. If you do not set a value, the client TLS is not enabled. The `clientTLS` parameter has the required subfields, `spec.clientTLS.clientCertificatePolicy` and `spec.clientTLS.ClientCA`.

Reviewer: Why a comma separated list of the required fields here vs. the bullets used for others?
Not a big deal, the inconsistency just jumped out out me. Maybe it's because the text looks like a copy of the OpenShift docs.
https://docs.redhat.com/en/documentation/openshift_container_platform/4.18/html/networking/networking-operators#nw-ingress-controller-configuration-parameters_configuring-ingress

> The `ClientCertificatePolicy` subfield accepts one of the two values: `Required` or `Optional`. Note that the ingress controller only checks client certificates for edge-terminated and reencrypt TLS routes; it cannot check certificates for cleartext HTTP or passthrough TLS routes. The `ClientCA` subfield specifies a config map that is in the openshift-ingress namespace. The config map should contain a CA certificate bundle. A config map is required for this field.

Reviewer 3: ```suggestion
The `ClientCertificatePolicy` subfield accepts one of the following two values: `Required` or `Optional`. Note that the ingress controller only checks client certificates for edge-terminated and reencrypt TLS routes; the ingress controller cannot check certificates for cleartext HTTP or passthrough TLS routes. The `ClientCA` subfield specifies a config map that is in the `openshift-ingress` namespace. The config map should contain a CA certificate bundle. A config map is required for this field.
```

> The `ClientCertificatePolicy` subfield accepts one of the two values: `Required` or `Optional`. Note that the ingress controller only checks client certificates for edge-terminated and reencrypt TLS routes; it cannot check certificates for cleartext HTTP or passthrough TLS routes. The `ClientCA` subfield specifies a config map that is in the openshift-ingress namespace. The config map should contain a CA certificate bundle. A config map is required for this field.

Reviewer 3: "it" is a vague referrent. Best to state the noun directly.

> The `AllowedSubjectPatterns` is an optional value that specifies a list of regular expressions, which are matched against the distinguished name on a valid client certificate to filter requests. The regular expressions must use PCRE syntax. This field must contain a valid expression or the MicroShift service will fail. At least one pattern must match a client certificate's distinguished name; otherwise, the ingress controller rejects the certificate and denies the connection. If you do not specify a value, the ingress controller does not reject certificates based on the distinguished name.

Reviewer 3: PCRE expand abbreviations.

Reviewer 3: MicroShift: does an attribute exist?

|`ingress.certificate.Secret`
|`string`
|The `certificateSecret` value is a reference to a secret that contains the default certificate that is served by the Ingress Controller. When Routes do not specify their own certificate, `certificateSecret` is used.

The secret must contain the following keys and data:
* `tls.crt`: certificate file contents
* `tls.key`: key file contents

If not set, a wildcard certificate is automatically generated and used. The certificate is valid for the Ingress Controller `domain` and `subdomains`, and
the generated certificate's CA is automatically integrated with the
cluster's truststore.

The in-use certificate, whether generated or user-specified, is automatically integrated with {product-title} built-in OAuth server.

|`ingress.clientTLS`
|`spec.clientTLS.clientCertificatePolicy`, `spec.clientTLS.ClientCA`, `AllowedSubjectPatterns`
|`clientTLS` authenticates client access to the cluster and services; as a result, mutual TLS authentication is enabled. If not set, then client TLS is not enabled. `clientTLS` has the required subfields, `spec.clientTLS.clientCertificatePolicy` and `spec.clientTLS.ClientCA`.

> The `ClientCertificatePolicy` subfield accepts one of the two values: `Required` or `Optional`. Note that the ingress controller only checks client certificates for edge-terminated and reencrypt TLS routes; it cannot check certificates for cleartext HTTP or passthrough TLS routes. The `ClientCA` subfield specifies a config map that is in the openshift-ingress namespace. The config map should contain a CA certificate bundle. A config map is required for this field.

Reviewer 4: Are you accepting this suggestion?

|`ingress.logEmptyRequests`
|`Log` or `Ignore`

> |The default value is `Log`. Specifies how connections on which empty requests are received are logged. These connections typically come from the health probes of a load balancer service health or a web browser's speculative connections, such as a `preconnect`. Logging typical requests might be undesirable, but requests can also be caused by network errors or port scans, in which case logging can be useful for diagnosing errors and detecting intrusion attempts.

Reviewer: There are other instances, one in this PR above, where "Default" vs. "The default" is used. If you're correcting it here, should it be make consistent throughout the doc?

"health probes of a load balancer service health" <-- drop the second use of "health".

|`ingress.routeAdmissionPolicy.wildcardPolicy`
|`WildcardsAllowed`, `WildcardsDisallowed`
|Describes how routes with wildcard policies are handled by the Ingress Controller.

* `WildcardsAllowed`: Indicates routes with any wildcard policy are admitted by the Ingress Controller.

* `WildcardsDisallowed`: Indicates only routes with a wildcard policy of `None` are admitted by the Ingress Controller. Updating `wildcardPolicy` from `WildcardsAllowed` to `WildcardsDisallowed` causes admitted routes with a wildcard policy of `Subdomain` to stop working. These routes must be recreated to a wildcard policy of `None` to be readmitted by the Ingress Controller. `WildcardsDisallowed` is the default setting.

|`ingress.status`
|`Managed` or `Removed`
|Router status. Default is `Managed`.

|`ingress.tlsSecurityProfile`
|`object`
|Specifies settings for TLS connections for Ingress Controllers. If you do not set one, the default value is based on the `apiservers.config.openshift.io/cluster` resource.

|`ingress.tlsSecurityProfile.type`
|`Old`, `Intermediate`, `Modern`, 'Custom' 
|Specifies the profile type for the TLS Security.

> When using the `Old`, `Intermediate`, and `Modern` profile types, the effective profile configuration is subject to change between releases. For example, given a specification to use the `Intermediate` profile deployed on release `X.Y.Z`, an upgrade to release `X.Y.Z+1` may cause a new profile configuration to be applied to the Ingress Controller, resulting in a rollout.

Reviewer 5: `Intermediate` is the current default (when nothing is provided), i think we should make this more clear.

|`443`
|Default port shown. Configurable. Valid value is a single, unique port in the `1-65535` range. The values of the `ports.http` and `ports.https` fields cannot be the same.

|`ingress.routeAdmissionPolicy`
|`Strict` or `InterNamespaceAllowed`
|Defines a policy for handling new route claims, such as allowing or denying claims across namespaces. By default, allows routes to claim different paths of the same hostname across namespaces. Specifying `Strict` prevents routes in different namespaces from claiming the same hostname. If the value is deleted in a customized {microshift-short} `config.yaml`, the `InterNamespaceAllowed` value is automatically set.

> |`ingress.routeAdmissionPolicy.namespaceOwnership`
> |`WildcardsAllowed`, `WildcardsDisallowed`
> |Describes how hostname claims across namespaces should be handled. The default value is `Strict`.
> 
> * `Strict`: does not allow routes to claim the same hostname across namespaces.
> * `InterNamespaceAllowed`: allows routes to claim different paths of the same hostname across namespaces.

Reviewer 4: Are these values correct here?  Should the `Strict` and `

```suggestion
|`ingress.routeAdmissionPolicy.namespaceOwnership`
|`Strict`, `InterNamespaceAllowed`
|Describes how hostname claims across namespaces should be handled. The default value is `Strict`.

* `Strict`: does not allow routes to claim the same hostname across namespaces.
* `InterNamespaceAllowed`: allows routes to claim different paths of the same hostname across namespaces.
```

> * `InterNamespaceAllowed`: allows routes to claim different paths of the same hostname across namespaces.

Author: Definitely awry. I mixed up the subfields `ingress.routeAdmissionPolicy.namespaceOwnership` and `ingress.routeAdmissionPolicy.wildcardPolicy`. Please review my rewording.

|`ingress.routeAdmissionPolicy.wildcardPolicy`
|`WildcardsAllowed`, `WildcardsDisallowed`
|Describes how routes with wildcard policies are handled by the ingress controller.

> * `WildcardsAllowed`: Indicates routes with any wildcard policy are admitted by the ingress controller.

Reviewer 3: I see "ingress controller" and "Ingress Controller" versions. Best to stick to one.

* `WildcardsDisallowed`: Indicates only routes with a wildcard policy of `None` are admitted by the Ingress Controller. Updating `wildcardPolicy` from `WildcardsAllowed` to `WildcardsDisallowed` causes admitted routes with a wildcard policy of `Subdomain` to stop working. These routes must be recreated to a wildcard policy of `None` to be readmitted by the Ingress Controller. `WildcardsDisallowed` is the default setting.

|`ingress.status`
|`Managed` or `Removed`
|Router status. Default is `Managed`.

|`ingress.tlsSecurityProfile`
|`object`
|Specifies settings for TLS connections for ingress controllers. If you do not set one, the default value is based on the `apiservers.config.openshift.io/cluster` resource.

|`ingress.tlsSecurityProfile.type`
|`Old`, `Intermediate`, `Modern`, 'Custom' 
|Specifies the profile type for the TLS Security. The default value is `Intermediate`.

> When using the `Old`, `Intermediate`, and `Modern` profile types, the effective profile configuration is subject to change between releases. For example, given a specification to use the `Intermediate` profile deployed on release `X.Y.Z`, an upgrade to release `X.Y.Z+1` may cause a new profile configuration to be applied to the ingress controller, resulting in a rollout.

Author: When I worked at SAS, the style guide asked us to replace "may" with "might" because "may" could be ambiguous to the translators.

|`443`
|Default port shown. Configurable. Valid value is a single, unique port in the `1-65535` range. The values of the `ports.http` and `ports.https` fields cannot be the same.

> //Has this parameter been renamed to 'ingress.routeAdmission'?

Reviewer 6: No, see https://github.com/openshift/microshift/blob/main/packaging/microshift/config.yaml#L170; this parameter is still present and should not be deleted unless there is an engineering PR inflight that changes it.

|`network.serviceNodePortRange`
|`range`

> |The port range allowed for Kubernetes services of type `NodePort`. If you do not specify the range, the default range of 30000-32767 is used. Services without a `NodePort` specified are automatically allocated one from this range. This parameter can be updated after {microshift-short} starts.

Reviewer 3: ```suggestion
|The port range allowed for Kubernetes services of type `NodePort`. If you do not specify the range, the default range of `30000-32767` is used. Services without a `NodePort` specified are automatically allocated one from this range. This parameter can be updated after {microshift-short} starts.
```

|`network.serviceNodePortRange`
|`range`

> |The port range allowed for Kubernetes services of type `NodePort`. If you do not specify the range, the default range of `30000-32767` is used. Services without a `NodePort` specified are automatically allocated one from this range. This parameter can be updated after {microshift-short} starts.

Reviewer: Can you double check with the SME about this statement?
"This parameter can be updated after {microshift-short} starts."
Is it __can__ or __can not__?

Author: Evgeny informed me that he is unaware of any restrictions for updating the parameter after MicroShift starts.
